src/components/shared/Button/index.tsx

```diff
@@ -1,12 +1,27 @@
 import React, { FC } from 'react';
 import { ButtonPropsType } from './type';
 
-export const Button: FC<ButtonPropsType> = ({ type, children, onClick }) => {
+export const Button: FC<ButtonPropsType> = ({
+  type,
+  children,
+  onClick,
+  background,
+  disabled,
+}) => {
   return (
     <button
+      disabled={disabled}
       type={type}
       onClick={onClick}
-      className="w-4/5 bg-lime-500 rounded-3xl uppercase text-green-50 p-2 hover:bg-lime-400"
+      className={
+        background
```

Comment:
if there's nested ternary operations i'd move them to a variable or if else statement and just put variable here instead.